rest-service/manager_rest/security/authentication.py

```diff
@@ -52,7 +52,10 @@ def authenticate(self, request):
                 and not self.token_based_auth:
             self.logger.debug('using external auth')
             user = user_handler.get_user_from_auth(request.authorization)
-            user = self.external_auth.authenticate(request, user)
+            response = self.external_auth.authenticate(request, user)
+            if type(response) == Response:
```

Comment:
use `isinstance(response, Response)` so that subclasses of Response can also pass this